### DISCUSSION TITLE
How to use fcal for my own tracking and ultrasound devices?

### DISCUSSION THREAD

**Discussion by 1939938853** [PlusToolkit/PlusLib#928] [Q&A]
Hello, 

PlusToolkit looks great. But my  tracking and ultrasound devices are not included in it. How to add my own devices to PlusToolkit fcal? Any guideline for the interfacing? Is it in PlusLib?

Thank you!

**Comment by adamrankin** [comment]
If you want to add a new device class (new tracker or imaging device with an API/SDK), you can follow this guide to add a new device:

https://plustoolkit.github.io/devicecode

**Reply by 1939938853** [reply]
Hello Adam,

For new ultrasound devices, should I follow the same steps/templates as the tracking device? Which folder in PlusLib\src\PlusDataCollection contains a good example for new ultrasound device interfacing?

Thank you!
